MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -538,6 +577,373 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputHermiteSplineModel(vtkMRMLMarkups
+    return;
+  }
+
+  int numPoints = points->GetNumberOfPoints();
+  // redundant error checking, to be safe
+  if (numPoints < LINE_MIN_NUMBER_POINTS)
+  {
+    vtkErrorMacro("Not enough points to compute a polynomial fit. Need at least " << LINE_MIN_NUMBER_POINTS << " points but " << numPoints << " are provided. No output created.");
+    return;
+  }
+
+  // special case, fit a line. The polynomial solver does not work with only 2 points.
+  if (numPoints == LINE_MIN_NUMBER_POINTS)
+  {
+    vtkWarningMacro("Only " << LINE_MIN_NUMBER_POINTS << " provided. Fitting line.");
+    UpdateOutputLinearModel(markupsToModelModuleNode, markupsPointsPolyData, outputPolyData);
+    return;
+  }
+
+  // check size of point parameters array for consistency
+  if ( markupsPointsParameters->GetNumberOfTuples() != numPoints )
+  {
+    vtkErrorMacro("Incorrect number of point parameters provided. Should have " << numPoints << " parameters (one for each point), but " << markupsPointsParameters->GetNumberOfTuples() << " are provided. No output created.");
+    return;
+  }
+  
+  int numPolynomialCoefficients = markupsToModelModuleNode->GetPolynomialOrder() + 1;
+  // special case, if polynomial is underdetermined, change the order of the polynomial
+  std::set<double> uniquePointParameters;
+  for (int i = 0; i < numPoints; i++)
+    uniquePointParameters.insert(markupsPointsParameters->GetValue(i));
+  int numUniquePointParameters = uniquePointParameters.size();
+  if (numUniquePointParameters < numPolynomialCoefficients)
+  {
+    vtkWarningMacro("Not enough points to compute a polynomial fit. " << "For an order " << markupsToModelModuleNode->GetPolynomialOrder() << " polynomial, at least " << numPolynomialCoefficients << " points with unique parameters are needed. "
+                    << numUniquePointParameters << " points with unique parameters were found. "
+                    << "An order " << (numUniquePointParameters - 1) << " polynomial will be created instead.");
+    numPolynomialCoefficients = numUniquePointParameters;
+  }
+
+  // independent values (parameter along the curve)
+  int numIndependentValues = numPoints * numPolynomialCoefficients;
+  std::vector<double> independentValues(numIndependentValues); // independent values
+  independentValues.assign(numIndependentValues, 0.0);
+  for (int c = 0; c < numPolynomialCoefficients; c++) // o = degree index
+  {
+    for (int p = 0; p < numPoints; p++) // p = point index
+    {
+      double value = std::pow( markupsPointsParameters->GetValue(p), c );
+      independentValues[p * numPolynomialCoefficients + c] = value;
+    }
+  }
+  std::vector<double*> independentMatrix(numPoints);
+  independentMatrix.assign(numPoints, NULL);
+  for (int p = 0; p < numPoints; p++)
+  {
+    independentMatrix[p] = &(independentValues[p * numPolynomialCoefficients]);
+  }
+  double** independentMatrixPtr = &(independentMatrix[0]);
```

Comment:
move variable creation to the closest place it is first _used_ (I had to scroll through a lot to find where this variable is actually used and for what)